sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -324,84 +320,107 @@ def __init__(self, **kwargs: Any) -> None:
         self._on_refresh_success: Optional[Callable] = kwargs.pop("on_refresh_success", None)
         self._on_refresh_error: Optional[Callable[[Exception], None]] = kwargs.pop("on_refresh_error", None)
 
-    def refresh(self, **kwargs) -> None:  # pylint: disable=too-many-statements
+    def _configuration_refresh(self, client, headers, **kwargs) -> Tuple[bool, List[ConfigurationSetting]]:
+        configuration_settings: List[ConfigurationSetting] = []
+        refreshed_configs, self._refresh_on, configuration_settings = client.refresh_configuration_settings(
```

Comment:
That's quite odd that client.refresh_configuration_settings and client.refresh_feature_flags behave differently.

Reply:
I might be able to refactor this with the new logic, but as we get the configuration settings we check if any of them are watch keys and get the update etag values.